src/vcpkg/base/files.cpp

```diff
@@ -1682,6 +1824,206 @@ namespace vcpkg
         {
             return get_regular_files_impl<stdfs::directory_iterator>(dir, ec);
         }
+#else // ^^^ _WIN32 // !_WIN32 vvv
+#if defined(_DIRENT_HAVE_D_TYPE)
+#define VCPKG_GET_D_TYPE(dirent) ((dirent)->d_type)
+#define VCPKG_DT_UNKNOWN DT_UNKNOWN
+#define VCPKG_DT_REG DT_REG
+#define VCPKG_DT_DIR DT_DIR
+#else
+#define VCPKG_GET_D_TYPE(dirent) 0
+#define VCPKG_DT_UNKNOWN 0
+#define VCPKG_DT_REG 1
+#define VCPKG_DT_DIR 2
+#endif
+
+        // Selector is a function taking (unsigned char dtype, mode_t mode) and returning bool
+        // dtype is the value from the d_type member of struct dirent. If the system does not support the d_type
+        // member, this will always be set to VCPKG_DT_UNKNOWN.
+        // mode is the st_mode member of struct stat. It is populated if and only if dtype is VCPKG_DT_UNKNOWN.
+        // Note that many file systems always set dtype to DT_UNKNOWN, so the lstat fallback must exist even on systems
+        // which have the d_type member
+        template<class Selector>
+        static void get_files_recursive_impl(std::vector<Path>& result,
+                                             const Path& base,
+                                             std::error_code& ec,
+                                             Selector selector)
+        {
+            ReadDirOp op{base, ec};
+            if (ec)
+            {
+                translate_not_found_to_success(ec);
+            }
+            else
+            {
+                for (;;)
+                {
+                    errno = 0;
+                    auto entry = op.read();
+                    if (!entry)
+                    {
+                        if (errno != 0)
+                        {
+                            ec.assign(errno, std::generic_category());
+                            return;
+                        }
+
+                        // no more entries left
+                        break;
+                    }
+
+                    if (is_dot_or_dot_dot(entry->d_name))
+                    {
+                        continue;
+                    }
+
+                    auto full = base / entry->d_name;
```

Comment:
```suggestion
                    const auto full = base / entry->d_name;
```
or hoist the variable out of the loop to reuse the buffer